components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -865,6 +865,22 @@ public void removeOAuthApplicationData(String consumerKey) throws IdentityOAuthA
 
     }
 
+    /**
+     * Remove all OAuth consumer applications of a tenant.
+     *
+     * @param tenantId Id of the tenant
+     * @throws IdentityOAuthAdminException
+     */
+    public void removeAllOAuthApplicationData(int tenantId) throws IdentityOAuthAdminException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Deleting all OAuth Application data of the tenant: " + tenantId);
+        }
+
+        OAuthAppDAO dao = new OAuthAppDAO();
+        dao.removeConsumerApplicationsByTenantId(tenantId);
```

Comment:
What about the applications cached?

Reply:
Also, we need to check what happens to the OAuth app -> Service Provider associations with this delete.

Reply:
Do we need to separately delete OAuth apps?  OAuth apps should be auto-deleted when we delete the Service Providers. So, This step is not required. 

@mefarazath WDYT?

Reply:
I think, there we only delete from SP tables. Thus let's keep this.  As Farasath mentioned, we have to delete the associations too.